script/lint.js

```diff
@@ -136,12 +137,54 @@ const LINTERS = [ {
       process.exit(1)
     }
   }
+}, {
+  key: 'patches',
+  roots: ['patches'],
+  test: () => true,
+  run: () => {
+    const patchesDir = path.resolve(__dirname, '../patches')
+    for (const patchTarget of fs.readdirSync(patchesDir)) {
+      const targetDir = path.resolve(patchesDir, patchTarget)
+      // If the config does not exist that is OK, we just skip this dir
+      const targetConfig = path.resolve(targetDir, 'config.json')
+      if (!fs.existsSync(targetConfig)) continue
+
+      const config = JSON.parse(fs.readFileSync(targetConfig, 'utf8'))
+      for (const key of Object.keys(config)) {
+        // The directory the config points to should exist
+        const targetPatchesDir = path.resolve(__dirname, '../../..', key)
+        expect(fs.existsSync(targetPatchesDir)).to.equal(true, `target patch directory: "${targetPatchesDir}" should exist`)
+        // We need a .patches file
+        const dotPatchesPath = path.resolve(targetPatchesDir, '.patches')
+        expect(fs.existsSync(dotPatchesPath)).to.equal(true, `.patches file: "${dotPatchesPath}" should exist`)
+
+        // Read the patch list
+        const patchFileList = fs.readFileSync(dotPatchesPath, 'utf8').trim().split('\n')
+        const patchFileSet = new Set(patchFileList)
+        expect(patchFileList).to.have.length(patchFileSet.size, 'each patch file should only be in the .patches file once')
+        for (const file of fs.readdirSync(targetPatchesDir)) {
+          // Ignore the .patches file and READMEs
+          if (file === '.patches' || file === 'README.md') continue
```

Comment:
maybe only look at `.patch` files? `git-export-patches` should only ever produce files ending in `.patch`.

Reply:
Yeah, which is why I wanted to yell at people if a random file ended up in that directory (it should never be there)

Reply:
Hm, idk. Extra files don't hurt; `git-export-patch` only cares about `.patch` files and the `.patches` list. I can imagine it being fine to have other files in there? But I suppose it doesn't really matter. Up to you.